docs/source/etl/variant-data.rst

```diff
@@ -118,6 +118,14 @@ For both the single and sharded VCF writer, you can use the following option to
 |                 |        |             | this file is used as the VCF header for each partition.                                                            |
 +-----------------+--------+-------------+--------------------------------------------------------------------------------------------------------------------+
 
+.. _infer-vcf-samples:
+
+If the header is inferred from the DataFrame, the sample IDs are derived from the rows. If there are missing sample IDs,
+they will be represented as ``sample_n``, for which ``n`` reflects the index of the sample in a row.
+
+- For the single VCF writer, the inferred sample IDs are the distinct set of all sample IDs from the DataFrame.
+- For the sharded VCF writer, the sample IDs are inferred from the first row of each partition. If the rows do not
```

Comment:
say that it's an error if every row doesn't have the same sample ids or the same number of samples if there are no sample ids